Sources/SnapshotTesting/Snapshotting/NSImage.swift

```diff
@@ -77,12 +77,17 @@ private func compare(_ old: NSImage, _ new: NSImage, precision: Float) -> Bool {
   let newRep = NSBitmapImageRep(cgImage: newerCgImage)
   var differentPixelCount = 0
   let pixelCount = oldRep.pixelsWide * oldRep.pixelsHigh
-  let threshold = 1 - precision
-  for x in 0..<oldRep.pixelsWide {
-    for y in 0..<oldRep.pixelsHigh {
-      if oldRep.colorAt(x: x, y: y) != newRep.colorAt(x: x, y: y) { differentPixelCount += 1 }
-      if Float(differentPixelCount) / Float(pixelCount) > threshold { return false}
+  let threshold = (1 - precision) * Float(pixelCount)
+  let p1: UnsafeMutablePointer<UInt8> = oldRep.bitmapData!
+  let p2: UnsafeMutablePointer<UInt8> = newRep.bitmapData!
+  for offset in 0 ..< pixelCount {
+    if p1[offset] != p2[offset]
+        || p1[offset + 1] != p2[offset + 1]
+        || p1[offset + 2] != p2[offset + 2]
+        || p1[offset + 3] != p2[offset + 3] {
+        differentPixelCount += 1
     }
```

Comment:
Mmmm, I've re-read my PR just now I think this loop is doing unnecessary extra work.

Shouldn't it advance by 4 on each loop? The four fields are R, G, B, and alpha, each 8 bit values. By the looks of it I'm comparing

0 - 0
1 - 1
2 - 2
3 - 3

and then on next iteration

1 - 1
2 - 2
3 - 3
4 - 4

instead of

5 - 5
6 - 6
...

Or maybe I'm just confused now 😅

Reply:
Yeah indeed, it seems like it right? Not sure about the type and what the offset should be, but if this is true aren't you then also only checking 1/4 of all the pixels? 
So you should either calculate an offset within the loop doing:
`let actualOffset = offset * 4`
and then checking the four channels. Or just multiply the number of pixels with the amount of channels.

Reply:
Ah huh, that makes sense. Wanna do a follow-up PR?

An alternative idea for optimization in the future would be to divide the total bytes into precision-bucketed chunks and use `memcmp` for each slice—I'm not sure this is feasible, though, and I'm not sure how often it would be faster vs. slower compared with your approach.

Reply:
I'll double check this asap but I think @JaapWijnen is right - this is doing overlapping checks of the first quarter of the pixels. It might be as simple as doing `offset * 4 + 0`, `offset * 4 + 1`, ... inside the loop.

I'll prep a sample image with the last pixel different to confirm.

Reply:
@finestructure sorry I was kind of intrigued so went ahead and made the PR sorry! Missed these comments.